src/components/views/SupplierLookup.tsx

```diff
@@ -0,0 +1,184 @@
+  const [selectedSupplierIndex, setSelectedSupplierIndex] = useState<
+    number | null
+  >(null);
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setName(event.target.value);
+  };
+
+  const handleIndexChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSIndex(event.target.value ? parseInt(event.target.value) : null);
+  };
+
+  const handleCityChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCity(event.target.value);
+  };
+
+  const handleSearch = (event: React.FormEvent) => {
+    event.preventDefault();
+    const hardcodedSuppliers: Supplier[] = [
+      { name: 'Andemis', supplierIndex: 1, city: 'San Francisco' },
+      { name: 'Anna', supplierIndex: 2, city: 'Machu Picchu' },
+      { name: 'Mathew', supplierIndex: 3, city: 'Kigali' },
+    ];
+
+    const filtered = hardcodedSuppliers.filter((supplier) => {
+      return (
+        (!name || supplier.name.toLowerCase().includes(name.toLowerCase())) &&
+        (supplierIndex === null || supplier.supplierIndex === supplierIndex) &&
+        (!city ||
+          (supplier.city &&
+            supplier.city.toLowerCase().includes(city.toLowerCase())))
+      );
+    });
+
+    setFilteredSuppliers(filtered);
+  };
+
+  const handleReset = () => {
+    setName('');
+    setSIndex(null);
+    setCity('');
+    setFilteredSuppliers([]);
+    setSelectedSupplierIndex(null);
+  };
+
+  const handleClose = () => {
+    if (onClose) onClose();
+  };
+
+  const handleSelectSupplier = (index: number) => {
+    setSelectedSupplierIndex(index);
+  };
+
+  const handleSubmit = () => {
+    if (selectedSupplierIndex !== null) {
+      const selectedSupplier = filteredSuppliers[selectedSupplierIndex];
+      onSupplierSelect(selectedSupplier);
+      handleClose();
+    }
+  };
+
+  return (
+    <dialog open>
+      <div className="modal-backdrop">
+        <form
+          className="supplier-container"
+          onSubmit={handleSearch}
+        >
+          <div className="top-bar">
+            <h2 className="top-bar-title">Search for suppliers</h2>
+
+            <div
+              className="x-btn"
+              onClick={onClose}
+            >
+              X
+            </div>
+          </div>
+          <div className="search-supplier-inputs-container">
+            <div className="top-bar-title-container">
+              <div className="top-bar-title">▼ Search criteria</div>
+            </div>
+            <div className="form-inputs">
+              <div className="inputs-container-supplier">
+                <div className="input-container">
+                  <label className="input-label">Supplier name</label>
+                  <Textfield
+                    name="supplierName"
+                    type="text"
+                    value={name}
+                    onChange={handleNameChange}
+                    placeholder="ANDEMIS"
```

Comment:
pleae remove placeholder. That's not needed. in general placeholder cannot be real input value (like here actual name of supplier) but it make sense to be something like a label. IN this case you don't need it